### DISCUSSION TITLE
Increase bus factor to at least 3 among all Mesa components

### DISCUSSION THREAD

**Discussion by EwoutH** [mesa/mesa#1962] [General]
We all know this famous excerpt:

> ## If Guido was hit by a bus?
> Michael McLay (_mclay@eeel.nist.gov_)
> _Wed, 29 Jun 94 10:07:42 EDT_
>
> What if you saw this posted tommorrow.
> > Guido's unexpected death has come as a shock to us all. Disgruntled
 members of the Tcl mob are suspected, but no smoking gun has been found...
>
> I just returned from a meeting in which the major objection to using
 Python was its dependence on Guido. They wanted to know if Python
 would survive if Guido disappeared. This is an important issue for
 businesses that may be considering the use of Python in a product.

I have currently noted that to some infrastructure components supporting Mesa only one or two maintainers of Mesa have access to it. That slows down development velocity, puts an unevenly large burden on some maintainers but most importantly poses a risk to Mesa continuity. Some infrastructure is more important that others, but PyPI is among one if not the most important.

The PyPI case shows this clearly: A failed 2.2.0 release (including announcement to 100+ users of a release that was not on PyPI yet), because a (for me) 5 minute job was left open for over a year (https://github.com/projectmesa/mesa/pull/1479, https://github.com/projectmesa/mesa/pull/1855). No hard feelings here but it's a weakness we should address.

Therefore I would propose that we make it a policy that to any Mesa component, at least 3 maintainers have full access. And if there isn't a clear reason why not, all maintainers should have access.

Further, we should increase the bus factor ReadtheDocs, Coveralls, GitHub Actions, Dependabot, Pre-commit.ci. And we should check if there isn't any (infrastructure) component we're missing.

We probably want to keep some internal (non-public facing) documentation on who has access to what.

**Comment by rht** [comment]
A test of the efficiency would be how fast it would take to release 2.2.1, which needs to contain #1960.

**Reply by jackiekazil** [reply]
Seeing we are up to 2.2.4 two weeks later makes me feels good. Because that means we definitely had a speed up.

**Comment by jackiekazil** [comment]
RE: #2009 & Bus factor... some history... 
Initially the elevated privileges sat with myself and @dmasad. As @dmasad  went on to do other amazing things in the world, I invited @tpike3 to lower that bus factor. Until I consciously say I so I will intended to always be around, but I am not able to do everything.  I know @dmasad and @tpike3 personally and knew more about their background, goals, etc. @tpike3 was the first to added. 

I support adding an additional person across all items, but how do we determine who that person in?  I can see different people for different reasons. This might be a simple exercise, but I want to make sure we have some rigor/thought behind it if multiple people are interested in this type of role.

**Comment by jackiekazil** [comment]
Also I would like to win the lottery, not be hit by a bus.

**Reply by EwoutH** [reply]
I'm rooting for you!

**Comment by jackiekazil** [comment]
One other piece of history -- once upon a time, we had three dedicated contributors all consistent and regular -- making contributions, attending meetings, being responsive. Then in one month -- two had a baby and the other bought a new house and moved and the project came to an absolute halt for maybe three months. So while we can reduce the lottery factor, it still doesn't mean that perfect storms can't happen.
